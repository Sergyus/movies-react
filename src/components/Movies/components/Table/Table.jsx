import React from 'react';
import TableCardContainer from './components/TableCard';

const Table = ({ movies, toggleModal }) => (movies.map(movie => (
  <TableCardContainer
    key={movie.id}
    movie={movie}
    toggleModal={toggleModal}
  />
))
);

export default Table;
