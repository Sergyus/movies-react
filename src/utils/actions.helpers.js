export const createAsyncFnRequest = requestName => ({
  type: requestName,
});

export const createAsyncFnFailure = (requestName, error) => ({
  type: requestName,
  error: error.message,
});
