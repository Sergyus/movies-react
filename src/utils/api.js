const MOVIE_DB_API_KEY = "ed98d25919802b5020e8feb6c3923886";
const MOVIE_DB_BASE_URL = "https://api.themoviedb.org/3";
const LANGUAGE = "ru-RU";

const createMovieDbUrl = (relativeUrl, queryParams) => {
  let baseUrl = `${MOVIE_DB_BASE_URL}${relativeUrl}?api_key=${MOVIE_DB_API_KEY}&language=${LANGUAGE}`;
  if (queryParams) {
    Object.keys(queryParams).forEach(paramName => {
      baseUrl += `&${paramName}=${queryParams[paramName]}`;
      return baseUrl;
    });
  }
  return baseUrl;
};

const getLatestMovies = ({ page }) => {
  const fullUrl = createMovieDbUrl("/movie/popular", {
    page
  });
  return fetch(fullUrl);
};

const getTopMovies = ({ page }) => {
  const fullUrl = createMovieDbUrl("/movie/top_rated", {
    page
  });
  return fetch(fullUrl);
};

const getUpcomingMovies = ({ page }) => {
  const fullUrl = createMovieDbUrl("/movie/upcoming", {
    page
  });
  return fetch(fullUrl);
};

const searchMovies = ({ page, query }) => {
  const fullUrl = `${createMovieDbUrl("/search/movie", {
    page,
    query
  })}&include_adult=false`;
  return fetch(fullUrl);
};

const getGenres = () => {
  const fullUrl = createMovieDbUrl("/genre/movie/list");
  return fetch(fullUrl);
};

const getMoviesByGenre = ({ page, genre }) => {
  const fullUrl = `${createMovieDbUrl("/discover/movie", {
    page
  })}&sort_by=popularity.desc&include_adult=false&include_video=false&with_genres=${genre}`;
  return fetch(fullUrl);
};

const getMovieDetails = ({ movieId }) => {
  const fullUrl = createMovieDbUrl(`/movie/${movieId}`);
  return fetch(fullUrl);
};

const getMovieVideo = movieId => {
  const fullUrl = createMovieDbUrl(`/movie/${movieId}/videos`);
  return fetch(fullUrl);
};

export {
  getTopMovies,
  searchMovies,
  getMovieDetails,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo
};
