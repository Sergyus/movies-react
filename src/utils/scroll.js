const getScrollDownPercentage = ({ scrollHeight, scrollPos, clientHeight }) => {
  const currentPosition = scrollPos + clientHeight;
  const percentageScrolled = currentPosition / scrollHeight;
  return percentageScrolled;
};

export default getScrollDownPercentage;
