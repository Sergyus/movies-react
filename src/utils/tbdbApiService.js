import noAvailableImage from '../assets/images/noAvailableImage.png';

const urlBase = 'https://image.tmdb.org/t/p/original';
const urlMinImg = 'https://image.tmdb.org/t/p/w300';

function getNormalRuntime(time) {
  const hours = Math.floor(time / 60);
  const minutes = time - 60 * hours;
  return `${hours}h ${minutes}m`;
}

export function getMovieGenres(genresIds = [], genres = []) {
  const genresArr = [];
  genresIds.map(genre => genres.map((movie) => {
    if (movie.id === genre) {
      return genresArr.push({ id: movie.id, name: movie.name });
    }
    return 0;
  }));
  return genresArr;
}

export function getMovieInfo(movie) {
  return {
    title: movie.title,
    genres: movie.genres,
    rating: movie.vote_average,
    runtime: getNormalRuntime(movie.runtime),
    overview: movie.overview,
    image: movie.backdrop_path ? `${urlBase}${movie.backdrop_path}` : noAvailableImage,
  };
}

export function getMovies(movies = [], allGenres = []) {
  return movies.map(movie => ({
    id: movie.id,
    rating: movie.vote_average,
    title: movie.title,
    genres: getMovieGenres(movie.genre_ids, allGenres),
    overview: movie.overview,
    image: movie.poster_path ? `${urlMinImg}${movie.poster_path}` : noAvailableImage,
  }));
}
