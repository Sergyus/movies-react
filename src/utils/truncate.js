const truncate = (description = '', maxLength) => (
  description.length && description.length > maxLength ? `${description.slice(0, maxLength)}...` : description
);

export default truncate;
