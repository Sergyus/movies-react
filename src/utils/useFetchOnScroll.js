import { useLayoutEffect } from 'react';
import getScrollDownPercentage from './scroll';

export const handleScroll = (
  isLoading, hasMorePages, filter, currentPage,
  getLatestMovies, getTopRatedMovies, getUpcomingMovies,
  getMoviesByGenre, currentGenre, getSearchMovies, currentSearchQuery, window,
) => {
  if (!isLoading) {
    const { scrollHeight, clientHeight } = window.document.documentElement;
    const scrollPos = window.pageYOffset;
    const percentageScrolled = getScrollDownPercentage(
      { scrollHeight, clientHeight, scrollPos },
    );
    if (percentageScrolled > 0.9 && hasMorePages) {
      switch (filter) {
        case 'Trending':
          getLatestMovies(currentPage + 1);
          break;
        case 'Top Rated':
          getTopRatedMovies(currentPage + 1);
          break;
        case 'Coming Soon':
          getUpcomingMovies(currentPage + 1);
          break;
        case 'Genres':
          getMoviesByGenre(currentPage + 1, currentGenre);
          break;
        case 'Search':
          getSearchMovies(currentPage + 1, currentSearchQuery);
          break;
        default: break;
      }
    }
  }
};
export const useFetchOnScroll = (...args) => {
  useLayoutEffect(() => {
    const callback = () => handleScroll(...args, window);
    document.addEventListener('scroll', callback);
    return () => {
      document.removeEventListener('scroll', callback);
    };
  }, [...args]);
};
