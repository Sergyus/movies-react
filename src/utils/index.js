import {
  getTopMovies, searchMovies, getGenres, getMoviesByGenre,
  getLatestMovies, getUpcomingMovies, getMovieVideo,
} from './api';
import { getMovies } from './tbdbApiService';
import truncate from './truncate';
import { useFetchOnScroll } from './useFetchOnScroll';
import useBodyScrollLock from './useBodyScrollLock';
import useOnClickOutside from './useOnClickOutside';
import { createAsyncFnFailure, createAsyncFnRequest } from './actions.helpers';

export {
  getTopMovies, searchMovies, getGenres, getMoviesByGenre,
  getLatestMovies, getUpcomingMovies, getMovieVideo,
  truncate, useBodyScrollLock, useOnClickOutside, getMovies,
  createAsyncFnFailure, createAsyncFnRequest, useFetchOnScroll,
};
