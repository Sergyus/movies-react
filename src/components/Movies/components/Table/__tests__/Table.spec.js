import React from 'react';
import { create } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Table from '..';
import { getMovies } from '../../../../../utils';
import mockMovies from '../../../../../assets/json/popular.json';
import { genres } from '../../../../../assets/json/genres.json';

describe('Table component renders correctly', () => {
  const movies = getMovies(mockMovies.results, genres);
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    app: {
      movies: [],
      genres: [],
      isLoading: true,
      currentPage: 1,
      error: false,
      filter: 'Trending',
      currentGenre: '',
      currentSearchQuery: '',
      videoKey: '',
      currentTotalPages: 1,
    },
    ui: {
      layout: 'table',
    },
  };
  const store = mockStore(initialState);
  it('Table renders correctly', () => {
    const tree = create(
      <Provider store={store}>
        <Table movies={movies} />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
