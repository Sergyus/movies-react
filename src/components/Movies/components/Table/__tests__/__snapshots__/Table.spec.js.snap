// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders correctly Table renders correctly 1`] = `
Array [
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/AnLCqcH66Ecu8VFmXu3RofK4SMK.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        T-34
      </h2>
      <span
        className="genre"
      >
        War
      </span>
      <span
        className="genre"
      >
        Drama
      </span>
      <span
        className="genre"
      >
        Adventure
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <p
        className="overview"
      >
        1941 - WWii. the second lieutenant Nikolai ivushkin, commander of a t-34, engages in an unequal battle against the tank ace Klaus Jager in a battle near moscow. His mission is more of a suicide - to destroy a dozen german tanks, all by himself. that said, luck does favour the bold. He wins the battle, barely survives, but loses his tank and lands himself in captivity for three long years... there was little to no chance for ivushkin and Jager to meet again, but the war knows how to throw a curve...
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          4.5
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/svIDTNUoajS8dLEo7EosxvyAsgJ.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Glass
      </h2>
      <span
        className="genre"
      >
        Thriller
      </span>
      <span
        className="genre"
      >
        Mystery
      </span>
      <span
        className="genre"
      >
        Drama
      </span>
      <p
        className="overview"
      >
        In a series of escalating encounters, security guard David Dunn uses his supernatural abilities to track Kevin Wendell Crumb, a disturbed man who has twenty-four personalities. Meanwhile, the shadowy presence of Elijah Price emerges as an orchestrator who holds secrets critical to both men.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.8
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Robin Hood
      </h2>
      <span
        className="genre"
      >
        Adventure
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <span
        className="genre"
      >
        Thriller
      </span>
      <p
        className="overview"
      >
        A war-hardened Crusader and his Moorish commander mount an audacious revolt against the corrupt English crown.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          5.8
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Bohemian Rhapsody
      </h2>
      <span
        className="genre"
      >
        Drama
      </span>
      <span
        className="genre"
      >
        Music
      </span>
      <p
        className="overview"
      >
        Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock 'n' roll band Queen in 1970. Hit songs become instant classics. When Mercury's increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet â€“ finding a way to keep the band together amid the success and excess.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          8.2
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/l76Rgp32z2UxjULApxGXAPpYdAP.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Overlord
      </h2>
      <span
        className="genre"
      >
        Horror
      </span>
      <span
        className="genre"
      >
        War
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <span
        className="genre"
      >
        Science Fiction
      </span>
      <p
        className="overview"
      >
        France, World War II, June 1944. On the eve of D-Day, some American paratroopers fall behind enemy lines after their plane crashes during a mission consisting of destroying a German radio tower in Cielblanc, a small village near the beaches of Normandy. After reaching their target, the surviving paratroopers realize that, in addition to fighting Nazi troops that patrol the village, they also must fight against something else.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.9
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/8yZAx7tlKRZIg7pJfaPhl00yHIQ.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Escape Room
      </h2>
      <span
        className="genre"
      >
        Horror
      </span>
      <span
        className="genre"
      >
        Thriller
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <p
        className="overview"
      >
        Six strangers find themselves in circumstances beyond their control, and must use their wits to survive.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.3
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Aquaman
      </h2>
      <span
        className="genre"
      >
        Action
      </span>
      <span
        className="genre"
      >
        Fantasy
      </span>
      <span
        className="genre"
      >
        Science Fiction
      </span>
      <span
        className="genre"
      >
        Adventure
      </span>
      <p
        className="overview"
      >
        Once home to the most advanced civilization on Earth, the city of Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people -- and then the surface world. Standing in his way is Aquaman, Orm's half-human, half-Atlantean brother and true heir to the throne. With help from royal counselor Vulko, Aquaman must retrieve the legendary Trident of Atlan and embrace his destiny as protector of the deep.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.8
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/uXJVpPXxZO4L8Rz3IG1Y8XvZJcg.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Mortal Engines
      </h2>
      <span
        className="genre"
      >
        Science Fiction
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <span
        className="genre"
      >
        Adventure
      </span>
      <span
        className="genre"
      >
        Fantasy
      </span>
      <span
        className="genre"
      >
        Thriller
      </span>
      <p
        className="overview"
      >
        Set in a world many thousands of years in the future. Earthâ€™s cities now roam the globe on huge wheels, devouring each other in a struggle for ever diminishing resources. On one of these massive Traction Cities, Tom Natsworthy has an unexpected encounter with a mysterious young woman from the Outlands who will change the course of his life forever.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/v3QyboWRoA4O9RbcsqH8tJMe8EB.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Creed II
      </h2>
      <span
        className="genre"
      >
        Drama
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <p
        className="overview"
      >
        Follows Adonis Creed's life inside and outside of the ring as he deals with new found fame, issues with his family, and his continuing quest to become a champion.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.7
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/ij0xoc13hGhrYIlXGGuPXWTh3vi.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        How to Train Your Dragon: The Hidden World
      </h2>
      <span
        className="genre"
      >
        Animation
      </span>
      <span
        className="genre"
      >
        Family
      </span>
      <span
        className="genre"
      >
        Adventure
      </span>
      <p
        className="overview"
      >
        As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothlessâ€™ discovery of an untamed, elusive mate draws the Night Fury away.  When danger mounts at home and Hiccupâ€™s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          8.1
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        A Star Is Born
      </h2>
      <span
        className="genre"
      >
        Drama
      </span>
      <span
        className="genre"
      >
        Music
      </span>
      <span
        className="genre"
      >
        Romance
      </span>
      <p
        className="overview"
      >
        Seasoned musician Jackson Maine discovers â€” and falls in love with â€” struggling artist Ally. She has just about given up on her dream to make it big as a singer â€” until Jack coaxes her into the spotlight. But even as Ally's career takes off, the personal side of their relationship is breaking down, as Jack fights an ongoing battle with his own internal demons.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          7.4
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/t0idiLMalKMj2pLsvqHrOM4LPdQ.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        The Mule
      </h2>
      <span
        className="genre"
      >
        Crime
      </span>
      <span
        className="genre"
      >
        Drama
      </span>
      <span
        className="genre"
      >
        Thriller
      </span>
      <p
        className="overview"
      >
        A 90 year old horticulturalist and Korean War veteran is caught transporting $3 million worth of cocaine through Illinois for a Mexican drug cartel.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.3
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Avengers: Infinity War
      </h2>
      <span
        className="genre"
      >
        Adventure
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <span
        className="genre"
      >
        Fantasy
      </span>
      <p
        className="overview"
      >
        As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          8.3
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Bumblebee
      </h2>
      <span
        className="genre"
      >
        Action
      </span>
      <span
        className="genre"
      >
        Adventure
      </span>
      <span
        className="genre"
      >
        Science Fiction
      </span>
      <p
        className="overview"
      >
        On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.5
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/f27RTnWEc2bhVtfSlKmxxypznfR.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Widows
      </h2>
      <span
        className="genre"
      >
        Crime
      </span>
      <span
        className="genre"
      >
        Drama
      </span>
      <span
        className="genre"
      >
        Thriller
      </span>
      <p
        className="overview"
      >
        A police shootout leaves four thieves dead during an explosive armed robbery attempt in Chicago. Their widows have nothing in common except a debt left behind by their spouses' criminal activities. Hoping to forge a future on their own terms, they join forces to pull off a heist.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.7
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/hgWAcic93phg4DOuQ8NrsgQWiqu.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Serenity
      </h2>
      <span
        className="genre"
      >
        Thriller
      </span>
      <span
        className="genre"
      >
        Drama
      </span>
      <p
        className="overview"
      >
        Baker Dill is a fishing boat captain leading tours off a tranquil, tropical enclave called Plymouth Island. His quiet life is shattered, however, when his ex-wife Karen tracks him down with a desperate plea for help.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          5.1
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Venom
      </h2>
      <span
        className="genre"
      >
        Science Fiction
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <p
        className="overview"
      >
        Investigative journalist Eddie Brock attempts a comeback following a scandal, but accidentally becomes the host of Venom, a violent, super powerful alien symbiote. Soon, he must rely on his newfound powers to protect the world from a shadowy organization looking for a symbiote of their own.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          6.6
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/hPuZR2bFoV0kuBrA98exWyvTf40.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Can You Ever Forgive Me?
      </h2>
      <span
        className="genre"
      >
        Drama
      </span>
      <span
        className="genre"
      >
        Comedy
      </span>
      <span
        className="genre"
      >
        Crime
      </span>
      <p
        className="overview"
      >
        When bestselling celebrity biographer Lee Israel is no longer able to get published because she has fallen out of step with current tastes, she turns her art form to deception, abetted by her loyal friend Jack.  It is a story of two difficult, socially inept, people somehow becoming partners-in-crime in forgeries: one creating and the other selling it.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          7.1
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/t0jwlwFUrtkPJoq5JH9YrRVROXZ.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        Fate/stay night: Heavenâ€™s Feel II. lost butterfly
      </h2>
      <span
        className="genre"
      >
        Animation
      </span>
      <span
        className="genre"
      >
        Action
      </span>
      <span
        className="genre"
      >
        Fantasy
      </span>
      <span
        className="genre"
      >
        Adventure
      </span>
      <p
        className="overview"
      >
        Theatrical-release adaptation of the visual novel "Fate/stay night", following the third and final route. (Part 2 of a trilogy.)
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          3.9
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
  <article
    className="card"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w300/nJVPgIq3icZMQ26p5uWIFdUZJfL.jpg"
    />
    <div
      className="info"
    >
      <h2
        className="title"
      >
        LelleBelle
      </h2>
      <span
        className="genre"
      >
        Music
      </span>
      <span
        className="genre"
      >
        Romance
      </span>
      <p
        className="overview"
      >
        19-year old Belle practices playing the violin diligently, but is unable to impart her music with a sensitive undertone. Her life changes when she discovers that sexual desire stimulates passion in her music.
      </p>
      <div
        className="rating"
      >
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <div
          className="star"
        />
        <span
          className="numRating"
        >
          4.4
        </span>
      </div>
      <button
        aria-label="Watch Now"
        className="btn primary"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
    </div>
  </article>,
]
`;
